Modules/LDAPObject.c

```diff
@@ -1172,9 +1172,8 @@ l_ldap_result4(LDAPObject *self, PyObject *args)
         else
             e = "ldap_parse_result";
         ldap_controls_free(serverctrls);
-        ldap_msgfree(msg);
```

Comment:
Since this `ldap_msgfree` is removed, it looks like `msg` will leak.

Reply:
It is now passed over to `LDAPraise_for_message` which calls `ldap_parse_result(..., 1)` in turn.